src/pages/components/modal/code.mdx

```diff
@@ -48,40 +86,425 @@ tabs: ["Usage", "Style", "Code", "Accessibility"]
+              </p>
+              <p>
+                Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aenean id accumsan augue. Phasellus consequat augue vitae tellus tincidunt posuere. Curabitur justo urna, consectetur vel elit iaculis, ultrices condimentum risus. Nulla facilisi. Etiam venenatis molestie tellus. Quisque consectetur non risus eu rutrum. 
+              </p>
+            </>
+          </p>
+        </ModalBody>
+        <ModalFooter primaryButtonText="OK" secondaryButtonText="Cancel" />
+      </ComposedModal>
+    )}
+  </ModalStateManager>
+</>
+  `}
+  </ComponentVariant>
 </ComponentDemo>
+
+### Opening/closing modal
+
+For both modal variants, you can open/close the modal by changing the `open` prop. For example, you can implement a launcher button with a React state and a `<Button>` that changes the state:
+
+```javascript
+function ModalStateManager() {
+  const [open, setOpen] = useState(false);
+  return (
+    <>
+      {typeof document === "undefined"
+        ? null
+        : ReactDOM.createPortal(
+            <ComposedModal open={open} onClose={() => setOpen(false)}>
+              ...
+            </ComposedModal>,
+            document.body
+          )}
+      <Button kind="primary" onClick={() => setOpen(true)}>
+        Open modal
+      </Button>
+    </>
+  );
+}
+```
+
+You can create an abstract version of such state manager, like:
+
+```javascript
+const ModalStateManager = ({
+  renderLauncher: LauncherContent,
+  children: ModalContent,
+}) => {
+  const [open, setOpen] = useState(false);
+  return (
+    <>
+      {!ModalContent || typeof document === "undefined"
+        ? null
+        : ReactDOM.createPortal(
+            <ModalContent open={open} setOpen={setOpen} />,
+            document.body
+          )}
+      {LauncherContent && <LauncherContent open={open} setOpen={setOpen} />}
+    </>
+  );
+};
+```
+
+<InlineNotification kind="warning">
+  <code>{`<Modal>`}</code>/<code>{`<ComposedModal>`}</code> has to be put at the
+  top-level in a React tree. The easiest way to ensure that is using React
+  portal as shown in above example.
+</InlineNotification>
+
+### Modal sizes
+
+There are four responsive <a href="https://www.carbondesignsystem.com/components/modal/usage/#modal-sizes">modal sizes</a>: `xs`, `small`, default, and `large`. You can set it via `size` prop:
+
+```javascript
+<ComposedModal size="lg">
+  <ModalHeader />
+  <ModalBody>
+    <p className="bx--modal-content__text">
+      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse
+      cursus fermentum risus, sit amet fringilla nunc pellentesque quis. Duis
+      quis odio ultrices, cursus lacus ac, posuere felis. Donec dignissim libero
+      in augue mattis, a molestie metus vestibulum. Aliquam placerat felis
+      ultrices lorem condimentum, nec ullamcorper felis porttitor.
+    </p>
+  </ModalBody>
+</ComposedModal>
+```
+
+Depending on the modal size of your choice and the viewport size,
```

Comment:
I think add an H4 here to help with scanability, call it "Alignment" so it matches with the [design usage](https://www.carbondesignsystem.com/components/modal/usage#alignment) maybe?